core/src/datayoga_core/expression.py

```diff
@@ -136,17 +124,25 @@ def exec_sql(self, data: List[Dict[str, Any]], expressions: Dict[str, str]) -> L
         """
         # use a CTE to create the in memory data structure
         data_inner = flatten_data(data)
-        cte_clause = self._get_cte(data_inner)
-
+        # builds a CTE expression for fetching in memory data
         column_names = data_inner[0].keys()
+        columns_clause = ','.join(f"[column{i+1}] as `{col}`" for i, col in enumerate(column_names))
 
-        # fetch the CTE and bind the variables
+        # values in the form of (?,?), (?,?)
+        values_clause_row = f"({','.join('?'*len(column_names))})"
+        values_clause = ','.join([values_clause_row]*len(data_inner))
```

Comment:
I'd add space before and after * but it might be a personal preference.

Reply:
done